packages/hint/src/lib/hint-context.ts

```diff
@@ -147,7 +147,7 @@ export class HintContext<E extends Events = Events> {
             location: position || { column: -1, line: -1 },
             message,
             resource,
-            severity: severity || this.severity,
+            severity: this.severity || severity,
```

Comment:
Am I misunderstanding or does this mean if a hint has a severity configured in `.hintrc` that it trumps any individual reports severity? If that's the case we at least need a way to specify `"default"` as a severity in the `.hintrc` file, though I think I'd prefer to put this back the way it was.

Reply:
I believe the way it is right now is not the proper way. The user has explicitly stated a severity error for a given hint and we should honor that. If they decide something is a warning and they get an error they will start scratching their heads (this has happened to me).

By really honoring the user's preference we can start adding default hint severity levels without breaking changes to those particular hints or the main package.
On top of that, the old and new schema can live together (with the expected user's behavior) while we transition everybody to the new schema. People will be able to upgrade smoothly without having to modify their hintrc file if they have one, get the alert message, and know what is coming down the road in case they want to start using the new format.

Reply:
Got it, but then don't we need to remove severity settings from our built-in configurations as part of this change or we'll always override any per-report severity?

Reply:
We will remove the severity from the config once we are ready and all the hints are updated.

(looks like GitHub doesn't answer in the thread when using email).